pandas/io/excel/_base.py

```diff
@@ -365,6 +346,7 @@ def read_excel(
     io,
     # sheet name is str or int -> DataFrame
     sheet_name: str | int = ...,
+    *,
```

Comment:
Very reasonable to do this, but at least worth a whatsnew note?

Reply:
This was deprecated before, but have to add whatsnews about the enforced deprecations in general